### ISSUE TITLE
GameManager shouldn't start ticking timers until the first timer is registered

### ISSUE THREAD

**Issue by jkomoros** [jkomoros/boardgame#599]
In NewGameManager we start a loop to tick timers every 250 ms, and there's no mechanism to stop that when or if the manager is no longer needed.

Most game managers don't need timers. And other game managers only need them in rare caess. It makes sense to only start ticking them the first time they're asked for, as a very simple optimization.

Test performance before and after the change; it could conceivably speed up tests non-trivially, because during them likely O(100) game managers are created, many of which don't need timers.

Related to #484, which would obviate the need for this minor improvement.
